daemon/delete.go

```diff
@@ -115,13 +115,9 @@ func (daemon *Daemon) rm(container *Container, forceRemove bool) (err error) {
 		}
 	}()
 
-	if _, err := daemon.containerGraphDB.Purge(container.ID); err != nil {
-		logrus.Debugf("Unable to remove container from link graph: %s", err)
-	}
-
 	metadata, err := daemon.layerStore.DeleteMount(container.ID)
 	layer.LogReleaseMetadata(metadata)
-	if err != nil {
+	if err != nil && err != layer.ErrMountDoesNotExist {
```

Comment:
@estesp This change fixed the container create cleanup problem.
Before this change I see the following error log when it fails to create a container:
```
Clean up Error! Cannot destroy container 9836efb819964e730e0fc32f7f3ee65b250900c2d6a1ed6cc20bbe1e9e62ef09: rmdriverfs: Driver aufs failed to remove root filesystem 9836efb819964e730e0fc32f7f3ee65b250900c2d6a1ed6cc20bbe1e9e62ef09: mount does not exist
```
Before this change each time it fails to create a container an orphan container `home` will be leaked in `/var/lib/docker/containers`.